winsup/utils/kill.cc

```diff
@@ -186,10 +187,20 @@ forcekill (pid_t pid, DWORD winpid, int sig, int wait)
       return;
     }
   if (!wait || WaitForSingleObject (h, 200) != WAIT_OBJECT_0)
-    if (sig && !TerminateProcess (h, sig << 8)
-	&& WaitForSingleObject (h, 200) != WAIT_OBJECT_0)
-      fprintf (stderr, "%s: couldn't kill pid %u, %u\n",
-	       prog_name, (unsigned int) dwpid, (unsigned int) GetLastError ());
+    {
+      HANDLE cur = GetCurrentProcess (), h2;
+      /* duplicate handle with access rights required for exit_process_tree() */
+      if (DuplicateHandle (cur, h, cur, &h2, PROCESS_CREATE_THREAD |
+			      PROCESS_QUERY_INFORMATION |
+			      PROCESS_VM_OPERATION |
+			      PROCESS_VM_WRITE | PROCESS_VM_READ |
+			      PROCESS_TERMINATE, FALSE, 0))
+      {
+        CloseHandle(h);
+        h = h2;
+      }
+      exit_process_tree (h2, 128 + sig);
```

Comment:
Would it be better to do
```suggestion
      exit_process_tree (h, 128 + sig);
```
in case DuplicateHandle failed?  I *think* h2 might be uninitialized in that case?

Reply:
Right, and the motivation was clearly to use `h` in any case. I _guess_ we can slip this into the PR, although it would make sense to backport this fixup to `msys2-3_2_0-release` first.

Reply:
this is in msys2-3_2_0-release and this PR now from what I see

Reply:
Sorry, I forgot to update this thread ;-)